exec/golang nil to []type error

Go+ code
```
func fn(data []int) {
    println(data)
}
fn(nil)
```
build exec/golang report error
```
[PANIC] github.com/goplus/gop/exec/golang/expr.go:238: Const: value type is unknown - []int
panic: Const: value type is unknown - []int
```

--- a/exec/golang/builder_test.go
+++ b/exec/golang/builder_test.go
@@ -21,6 +21,7 @@ import (
 	"go/ast"
 	"reflect"
 	"testing"
+	"unsafe"
 
 	"github.com/goplus/gop/cl"
 	"github.com/goplus/gop/exec.spec"
@@ -497,10 +498,21 @@ func TestReserved(t *testing.T) {
 
 func TestConstNil(t *testing.T) {
 	code := NewBuilder("main", nil, nil)
-	off := code.Reserve()
-	code.ReservedAsPush(off, []byte(nil))
-	if code.reserveds[off].Expr.(*ast.Ident).Name != "nil" {
-		t.Fatal("TestConstNil failed: reserveds is not set nil")
+	//Chan, Func, Map, Ptr, UnsafePointer, Interface, Slice
+	for _, v := range []interface{}{
+		(chan int)(nil),
+		(func())(nil),
+		map[int]string(nil),
+		(*struct{})(nil),
+		unsafe.Pointer(nil),
+		interface{}(nil),
+		[]byte(nil),
+	} {
+		off := code.Reserve()
+		code.ReservedAsPush(off, v)
+		if code.reserveds[off].Expr.(*ast.Ident).Name != "nil" {
+			t.Fatal("TestConstNil failed: reserveds is not set nil")
+		}
 	}
 }
 

--- a/exec/golang/builder_test.go
+++ b/exec/golang/builder_test.go
@@ -495,4 +495,13 @@ func TestReserved(t *testing.T) {
 	}
 }
 
+func TestConstNil(t *testing.T) {
+	code := NewBuilder("main", nil, nil)
+	off := code.Reserve()
+	code.ReservedAsPush(off, []byte(nil))
+	if code.reserveds[off].Expr.(*ast.Ident).Name != "nil" {
+		t.Fatal("TestConstNil failed: reserveds is not set nil")
+	}
+}
+
 // -----------------------------------------------------------------------------

--- a/exec/golang/expr.go
+++ b/exec/golang/expr.go
@@ -191,6 +191,11 @@ func Const(p *Builder, val interface{}) ast.Expr {
 	}
 	v := reflect.ValueOf(val)
 	kind := v.Kind()
+	if (kind >= reflect.Chan && kind <= reflect.Slice) || kind == reflect.UnsafePointer {
+		if v.IsNil() {
+			return nilIdent
+		}
+	}
 	if kind == reflect.String {
 		return StringConst(val.(string))
 	}

